tendermint/src/lite_impl/signed_header.rs

```diff
@@ -58,69 +59,107 @@ impl lite::Commit for block::signed_header::SignedHeader {
         }
 
         for commit_sig in self.commit.signatures.iter() {
-            // returns FaultyFullNode error if it detects a signer that is not present in the validator set
-            if let Some(val_addr) = commit_sig.validator_address {
-                if vals.validator(val_addr) == None {
-                    let reason = format!(
-                        "Found a faulty signer ({}) not present in the validator set ({})",
-                        val_addr,
-                        vals.hash()
-                    );
-                    fail!(Kind::FaultyFullNode, reason);
-                }
-            }
+            commit_sig.validate(vals)?;
         }
 
         Ok(())
     }
 }
 
-fn commit_to_votes(commit: block::Commit) -> Result<Vec<vote::Vote>, Error> {
+// this private helper function does *not* to any validation but extracts
+// all non-BlockIDFlagAbsent votes from the commit:
+fn non_absent_votes(commit: &block::Commit) -> Vec<vote::Vote> {
     let mut votes: Vec<vote::Vote> = Default::default();
     for (i, commit_sig) in commit.signatures.iter().enumerate() {
         if commit_sig.is_absent() {
             continue;
         }
 
-        match commit_sig.validator_address {
-            Some(val_addr) => {
-                if let Some(sig) = commit_sig.signature.clone() {
-                    let vote = vote::Vote {
-                        vote_type: vote::Type::Precommit,
-                        height: commit.height,
-                        round: commit.round,
-                        block_id: Option::from(commit.block_id.clone()),
-                        timestamp: commit_sig.timestamp,
-                        validator_address: val_addr,
-                        validator_index: i as u64,
-                        signature: sig,
-                    };
-                    votes.push(vote);
+        if let Some(val_addr) = commit_sig.validator_address {
+            if let Some(sig) = commit_sig.signature.clone() {
+                let vote = vote::Vote {
+                    vote_type: vote::Type::Precommit,
+                    height: commit.height,
+                    round: commit.round,
+                    block_id: Option::from(commit.block_id.clone()),
+                    timestamp: commit_sig.timestamp,
+                    validator_address: val_addr,
+                    validator_index: i as u64,
+                    signature: sig,
+                };
+                votes.push(vote);
+            }
+        }
+    }
+    votes
+}
+
+// TODO: consider moving this into commit_sig.rs instead and making it pub
+impl block::commit_sig::CommitSig {
```

Comment:
yes, I think this should be moved to commit_sig.rs

Reply:
Will do in a followup PR.